src/cargo/ops/lockfile.rs

```diff
@@ -68,10 +69,29 @@ pub fn write_lockfile(dst: &Path, resolve: &Resolve) -> CargoResult<()> {
         None => {}
     }
 
+    // Try to preserve CRLF line endings, see #2076
+    if has_crlf_line_endings(dst) {
```

Comment:
I've actually been meaning to implement a change where `Cargo.lock` isn't written if it's the same contents as before. This helps some situations where a source directory is mounted as read-only.

Along those lines, could this actually go ahead and do that? You'll need at least read one line here, and it's not much more to read the whole file, and then the guard against the `write` should be pretty easy to do!